src/Components/Web/src/JSComponents/JSComponentConfigurationExtensions.cs

```diff
@@ -7,19 +7,55 @@
 using Microsoft.AspNetCore.Components.Web.Infrastructure;
 using static Microsoft.AspNetCore.Internal.LinkerFlags;
 
-namespace Microsoft.AspNetCore.Builder
+namespace Microsoft.AspNetCore.Components.Web
 {
     /// <summary>
     /// Extension methods for working on an <see cref="IJSComponentConfiguration"/>.
     /// </summary>
     public static class JSComponentConfigurationExtensions
     {
+        // Having independent overloads for the cases with javaScriptInitializer and without it is needed for linkability,
+        // since calling the underlying .Add method with javaScriptInitializer is what causes the linker to retain code for
+        // the initializer feature.
+
         /// <summary>
         /// Marks the specified component type as allowed for instantiation from JavaScript.
         /// </summary>
         /// <typeparam name="TComponent">The component type.</typeparam>
-        [DynamicDependency(DynamicallyAccessedMemberTypes.All, typeof(JSComponentInterop))]
+        /// <param name="configuration">The <see cref="IJSComponentConfiguration"/>.</param>
+        /// <param name="identifier">A unique identifier for the component type that will be used by JavaScript code.</param>
         public static void RegisterForJavaScript<[DynamicallyAccessedMembers(Component)] TComponent>(this IJSComponentConfiguration configuration, string identifier) where TComponent : IComponent
-            => configuration.JSComponents.Add(identifier, typeof(TComponent));
+            => RegisterForJavaScript(configuration, typeof(TComponent), identifier);
+
+        /// <summary>
+        /// Marks the specified component type as allowed for instantiation from JavaScript.
+        /// </summary>
+        /// <typeparam name="TComponent">The component type.</typeparam>
+        /// <param name="configuration">The <see cref="IJSComponentConfiguration"/>.</param>
+        /// <param name="identifier">A unique identifier for the component type that will be used by JavaScript code.</param>
+        /// <param name="javaScriptInitializer">Specifies an optional identifier for a JavaScript function that will be called to register the custom element.</param>
```

Comment:
API review feedback: Remove the term `optional` from the docs since the parameter isn't really optional. Same for line  56